src/test/java/org/vafer/jdeb/DebMakerTestCase.java

```diff
@@ -232,4 +241,180 @@ public void visit(TarArchiveEntry entry, byte[] content) throws IOException {
+
+        final Map<String, TarArchiveEntry> filesInDeb = new HashMap<String, TarArchiveEntry>();
+
+        ArchiveWalker.walkData(deb, new ArchiveVisitor<TarArchiveEntry>() {
+            public void visit(TarArchiveEntry entry, byte[] content) throws IOException {
+                filesInDeb.put(entry.getName(), entry);
+            }
+        }, Compression.GZIP);
+
+        assertTrue("testfile wasn't found in the package", filesInDeb.containsKey("./test/testfile"));
+        assertTrue("testfile2 wasn't found in the package", filesInDeb.containsKey("./test/testfile2"));
+        assertTrue("testfile3 wasn't found in the package", filesInDeb.containsKey("./test/testfile3"));
+        assertTrue("testfile4 wasn't found in the package", filesInDeb.containsKey("./test/testfile4"));
+        assertTrue("/link/path-element.ext wasn't found in the package", filesInDeb.containsKey("./link/path-element.ext"));
+        assertEquals("/link/path-element.ext has wrong link target", "/link/target-element.ext", filesInDeb.get("./link/path-element.ext").getLinkName());
+
+        assertTrue("Cannot delete the file " + deb, deb.delete());
+    }
+
+    @Test
+    public void testLongLinkNameGnu() throws Exception {
+        String directoryName = "/var/log/" + longPathSuffix();
+        Mapper mapper = new PermMapper(-1, -1, "root", "root", -1, -1, 0, null);
+        String[] directories = { directoryName };
+        Mapper[] mappers = { mapper };
+        DataProducerPathTemplate dataProducer =  new DataProducerPathTemplate(directories, null, null, mappers);
+
+        DataProducer linkProducer = new DataProducerLink("/var/log/short", directoryName, true, null, null, null);
+
+        File deb = folder.newFile("file.deb");
+
+        List<DataProducer> producers = Arrays.asList(linkProducer, dataProducer);
+
+        DebMaker maker = new DebMaker(new NullConsole(), producers, null);
+        maker.setControl(new File(getClass().getResource("deb/control").toURI()));
+        maker.setTarLongFileMode("posix");
+        maker.setDeb(deb);
+        maker.setCompression(Compression.GZIP.toString());
+
+        BinaryPackageControlFile packageControlFile = maker.createDeb(Compression.GZIP);
+
+        assertTrue(packageControlFile.isValid());
+
+        final Map<String, TarArchiveEntry> filesInDeb = new HashMap<String, TarArchiveEntry>();
+
+        ArchiveWalker.walkData(deb, new ArchiveVisitor<TarArchiveEntry>() {
+            public void visit(TarArchiveEntry entry, byte[] content) throws IOException {
+                filesInDeb.put(entry.getName(), entry);
+            }
+        }, Compression.GZIP);
+
+        assertTrue("longname file wasn't found in the package", filesInDeb.containsKey(".//var/log/loooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongnameloooongname/"));
```

Comment:
The key should also use longPathSuffix() so that changing the longPath is one place.